Marlin/src/HAL/shared/backtrace/backtrace.cpp

```diff
@@ -36,7 +36,8 @@ static bool UnwReportOut(void *ctx, const UnwReport *bte) {
 
   const uint32_t a = bte->address, f = bte->function;
   MinSerial::TX('#');    MinSerial::TXDec(*p);    MinSerial::TX(" : ");
-  MinSerial::TX(bte->name?:"unknown");            MinSerial::TX('@');   MinSerial::TXHex(f);
+  //TODO: on HC32, calling MinSerial::TX(bte->name) causes a fault...
+  MinSerial::TX(TERN(TARGET_HC32F46x,, bte->name?:)"unknown");            MinSerial::TX('@');   MinSerial::TXHex(f);
```

Comment:
reverted, will create PR with fix in upstream since this probably affects not just HC32


```diff
diff --git a/Marlin/src/HAL/shared/backtrace/unwarmbytab.cpp b/Marlin/src/HAL/shared/backtrace/unwarmbytab.cpp
index 148927a19f..8594129aaf 100644
--- a/Marlin/src/HAL/shared/backtrace/unwarmbytab.cpp
+++ b/Marlin/src/HAL/shared/backtrace/unwarmbytab.cpp
@@ -55,7 +55,16 @@ static const char *UnwTabGetFunctionName(const UnwindCallbacks *cb, uint32_t add
     return nullptr;

   if ((flag_word & 0xFF000000) == 0xFF000000) {
-    return (const char *)(address - 4 - (flag_word & 0x00FFFFFF));
+    uint32_t fn_name_addr = address - 4 - (flag_word & 0x00FFFFFF);
+
+    // ensure the address is readable to avoid returning a bogus pointer
+    uint8_t dummy = 0;
+    if(!cb->readB(fn_name_addr, &dummy))
+    {
+      return nullptr;
+    }
+
+    return (const char *)fn_name_addr;
   }
   return nullptr;
 }
```

Reply:
created a PR in upstream with this change, + cherry-picked the commits into this PR